backend/src/main/java/com/zpi/backend/email/EmailService.java

```diff
@@ -0,0 +1,299 @@
+        context.setVariable("en_default_message",
+                translationService.getMessage("email.default-message", "en", null));
+        context.setVariable("en_button",
+                translationService.getMessage("email.button", "en", null));
+
+        context.setVariable("language", "pl");
+        context.setVariable("app_url", frontend_host + ":"+frontend_port);
+        context.setVariable("logo_image", getEncodedLogo());
+
+        return context;
+    }
+
+
+    @Async
+    public void sendEmailWithHtmlTemplate(User user, String subject, String templateName, Context context, EmailType emailType) {
+        try {
+            String htmlContent = templateEngine.process(templateName, context);
+            EmailLog log = new EmailLog(user,emailType, subject, htmlContent);
+            // Saving log
+            emailLogRepository.save(log);
+            // Sending mail
+            sendHTMLEmail(user.getEmail(), subject, htmlContent);
+            // Updating log
+            log.sent();
+            emailLogRepository.save(log);
+            logger.info(emailType.name() + " MAIL sent successfully");
+        } catch (Exception e) {
+            logger.error("Error while sending an email. "+e.getMessage());
+        }
+    }
+
+    private void sendHTMLEmail(String to, String subject, String htmlContent) throws MessagingException {
+        MimeMessage mimeMessage = emailSender.createMimeMessage();
+        MimeMessageHelper helper = new MimeMessageHelper(mimeMessage, "UTF-8");
+        helper.setTo(to);
+        helper.setSubject(subject);
+        helper.setText(htmlContent, true);
+        emailSender.send(mimeMessage);
+    }
+
+    private String getEncodedLogo() throws IOException {
+        byte[] bytes = IOUtils.toByteArray(logo_resource.getInputStream());
+        Base64.Encoder encoder = Base64.getEncoder();
+        String base64String = encoder.encodeToString(bytes);
+        return "data:image/png;base64," + base64String;
+    }
+
+    @Scheduled(cron = "0 */5 * ? * *", zone = "Europe/Warsaw")
+    public void sendUnsentEmails() {
+        List<EmailLog> unsentEmails = emailLogRepository.findAllBySentIsFalse();
+        logger.info("Checking e-mails to send.");
+        if (unsentEmails.size() > 0) {
+            for (EmailLog log : unsentEmails) {
+                try {
+                    sendHTMLEmail(
+                            log.getReceiver().getEmail(),
+                            log.getTitle(),
+                            log.getContent()
+                    );
+                    log.sent();
+                    emailLogRepository.save(log);
+                    logger.info("[LOG-"+log.getId()+"] "+log.getType().name() + " MAIL sent successfully");
+                } catch (Exception e) {
+                    logger.error("Error while sending an email. " + e.getMessage());
+                }
+            }
+        }
+    }
+
+    @Scheduled(cron = "0 0 8 ? * *", zone = "Europe/Warsaw")
```

Comment:
there will be some logic here when reservations are merged am i right? if so, is there a task to connect mailing service to reservations?

Reply:
that's right
<img width="694" alt="Screenshot 2023-11-19 at 21 17 39" src="https://github.com/zpi-project/gameshare-app/assets/91901843/e397cd00-d164-4bc5-a219-6c08d3dd645f">

Reply:
Probably they should have been named and estimated better, but ...